<html class="anndoc" data-anndoc-version="2.0" data-origid="18372165" id="18372165" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>18372165</title></head><body><article><section data-type="article"><div class="content"><p id="title">Molecular and clinical characterization in Japanese and Korean patients with Hailey-Hailey disease: six new mutations in the ATP2C1 gene.</p><p id="abstract">BACKGROUND: The autosomal dominant disorder Hailey-Hailey disease (HHD) results from mutations in the ATP2C1 gene, which encodes the human secretory pathway Ca2+/Mn2+ -ATPase protein 1. To date, over 90 pathological mutations scattered throughout ATP2C1 have been described with no indication of mutational hotspots or clustering of mutations. No paradigm for genotype-phenotype correlation has emerged. OBJECTIVES: To determine the pathogenic ATP2C1 abnormality in additional patients with HHD in order to provide further contributions to the understanding of the molecular basis of this disorder and to add the data to the known mutation database. METHODS: In this study, we investigated eight unrelated Japanese and Korean patients with HHD. We performed direct nucleotide sequencing of the ATP2C1 gene in all patients and RT-PCR analysis, using RNA extracted from a skin biopsy, in a patient with the mildest clinical features. RESULTS: We identified seven different heterozygous mutations in seven of the eight investigated patients, including three new single nucleotide deletion/duplication mutations: c.520delC; c.681dupA; c.956delC, three new donor splice site mutations: c.360+1G&gt;C; c.899+1G&gt;T; c.1570+2T&gt;C, as well as a previously described nonsense mutation: p.Arg153X. RT-PCR analysis in the mildest affected patient with a heterozygous c.360+1G&gt;C mutation, demonstrated expression of a short in-frame mutant transcript with exon 5 skipping, which may account for the mild phenotype. CONCLUSIONS: The results expand the known mutation spectrum in HHD and show the importance of RNA analysis for understanding the genotype-phenotype correlations more precisely.</p></div></section></article></body></html>