<html class="anndoc" data-anndoc-version="2.0" data-origid="12673796" id="12673796" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>12673796</title></head><body><article><section data-type="article"><div class="content"><p id="title">Detection of hotspot mutations and polymorphisms using an enhanced PCR-RFLP approach.</p><p id="abstract">Ethidium gel-based PCR-RFLP is widely used, and is perhaps the simplest method for detection of known mutations in cancer-related genes and for genotyping a wide range of other human diseases. However, its application is limited by the fact that it can only detect mutant alleles that are present in more than 5-10% of wild-type alleles. Here we present a method that allows a 1-2 order enhancement in the sensitivity of the widely used PCR-RFLP without substantially increasing the effort and cost associated with it. This method is a modification to our previously reported amplification via primer ligation at the mutation (APRIL-ATM) method, which utilizes ligation of a primer at a restriction site formed by a mutation, followed by a ligation-mediated PCR amplification which amplifies only the mutation-containing DNA molecules. By combining this method with the artificial introduction of restriction sites during PCR, we demonstrate that assays can be designed and validated for detecting hot-spot mutations in codons 273, 158, and 248 of the TP53 gene (p53) and potentially for most mutations of interest. This approach is validated by using samples where the mutation was artificially introduced at these p53 positions. The increased sensitivity offered by the method further allows us to rapidly screen for low frequency polymorphisms in pooled DNA samples. The frequency of an MSH2 missense polymorphism (965G&gt;A) was quantified in pooled genomic DNA samples from 205 and 221 U.S. and Polish colorectal cancer patients, respectively, and an equal number of ethnicity-matched controls. The data revealed a 3-5% prevalence of this polymorphism in the patient and the control populations. Individual sequencing of all 852 patient samples demonstrated an excellent agreement among the two independent approaches. The present enhanced PCR-RFLP reduces the effort involved in high throughput polymorphism studies and promises to find applications in genotyping and association studies involving low frequency polymorphisms and mutations.</p></div></section></article></body></html>