<html class="anndoc" data-anndoc-version="2.0" data-origid="15024732" id="15024732" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>15024732</title></head><body><article><section data-type="article"><div class="content"><p id="title">Analysis of the allele-specific expression of the mismatch repair gene MLH1 using a simple DHPLC-Based Method.</p><p id="abstract">Quantitative measures of allele-specific gene expression allow the indirect detection of mutations or sequence variants in regulatory elements or in other non-coding regions that may result in significant physiological or pathological changes of gene expression and may contribute to Mendelian or multifactorial disorders. We have devised a simple method, based on RT-PCR and single nucleotide primer extension (SNuPE) with unlabelled dideoxynucleotides, followed by DHPLC (denaturing high performance liquid chromatography). We established optimal conditions for separation of the extended products corresponding to the alleles of the c.655A&gt;G (p.Ile219Val) SNP, which is the most frequent exonic polymorphism of MLH1. We then genotyped 99 unrelated control subjects and measured the allele-specific MLH1 expression in the 40 heterozygous controls found in this group. This method allowed us to define a narrow range of normal biallelic expression of MLH1, each allele contributing between 44.7% and 55.3% of the total expression. We then measured the allele-specific expression in hereditary nonpolyposis colorectal cancer (HNPCC) patients with MLH1 mRNAs bearing different stop-codon or frame-shift mutations, or in-frame deletions, in order to detect the effects of nonsense-mediated mRNA decay (NMD). Defects that induce mRNA instability were identified unambiguously and the data were consistent with current models of NMD. This study provides a sensitive tool to identify indirectly MLH1 defects that may escape detection in genomic DNA screenings but result in a quantitative change at the mRNA level.</p></div></section></article></body></html>