<html class="anndoc" data-anndoc-version="2.0" data-origid="15759212" id="15759212" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>15759212</title></head><body><article><section data-type="article"><div class="content"><p id="title">Identifying candidate Hirschsprung disease-associated RET variants.</p><p id="abstract">Patients with sporadic Hirschsprung disease (HSCR) show increased allele sharing at markers in the 5' region of the RET locus, indicating the presence of a common ancestral RET mutation. In a previous study, we found a haplotype of six SNPs that was transmitted to 55.6% of our patients, whereas it was present in only 16.2% of the controls we used. Among the patients with that haplotype, 90.8% had it on both chromosomes, which led to a much higher risk of developing HSCR than when the haplotype occurred heterozygously. To more precisely define the HSCR-associated region and to identify candidate disease-associated variant(s), we sequenced the shared common haplotype region from 10 kb upstream of the RET gene through intron 1 and exon 2 (in total, 33 kb) in a patient homozygous for the common risk haplotype and in a control individual homozygous for the most common nonrisk haplotype. A comparison of these sequences revealed 86 sequence differences. Of these 86 variations, 8 proved to be in regions highly conserved among different vertebrates and within putative transcription factor binding sites. We therefore considered these as candidate disease-associated variants. Subsequent genotyping of these eight variants revealed a strong disease association for six of the eight markers. These six markers also showed the largest distortions in allele transmission. Interspecies comparison showed that only one of the six variations was located in a region also conserved in a nonmammalian species, making it the most likely candidate HSCR-associated variant.</p></div></section></article></body></html>