<html class="anndoc" data-anndoc-version="2.0" data-origid="16086324" id="16086324" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>16086324</title></head><body><article><section data-type="article"><div class="content"><p id="title">Comparison of yield and genotyping performance of multiple displacement amplification and OmniPlex whole genome amplified DNA generated from multiple DNA sources.</p><p id="abstract">The promise of whole genome amplification (WGA) is that genomic DNA (gDNA) quantity will not limit molecular genetic analyses. Multiple displacement amplification (MDA) and the OmniPlex PCR-based WGA protocols were evaluated using 4 and 5 ng of input gDNA from 60 gDNA samples from three tissue sources (mouthwash, buffy coat, and lymphoblast). WGA DNA (wgaDNA) yield and genotyping performance were evaluated using genotypes determined from gDNA and wgaDNA using the AmpFlSTR Identifiler assay and N = 49 TaqMan SNP assays. Short tandem repeat (STR) and SNP genotyping completion and concordance rates were significantly reduced with wgaDNA from all WGA methods compared with gDNA. OmniPlex wgaDNA exhibited a greater reduction in genotyping performance than MDA wgaDNA. Reduced wgaDNA genotyping performance was due to allelic (all protocols) and locus (OmniPlex) amplification bias leading to heterozygote and locus dropout, respectively, and %GC sequence content (%GC) was significantly correlated with TaqMan assay performance. Lymphoblast wgaDNA exhibited higher yield (OmniPlex), buffy coat wgaDNA exhibited higher STR genotyping completion (MDA), whereas mouthwash wgaDNA exhibited higher SNP genotyping discordance (MDA). Genotyping of wgaDNA generated from &lt; or = 5 ng gDNA, e.g., from archaeological, forensic, prenatal diagnostic, or pathology samples, may require additional genotyping validation with gDNA and/or more sophisticated analysis of genotypes incorporating observed reductions in genotyping performance.</p></div></section></article></body></html>