<html class="anndoc" data-anndoc-version="2.0" data-origid="16411177" id="16411177" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>16411177</title></head><body><article><section data-type="article"><div class="content"><p id="title">Masking selected sequence variation by incorporating mismatches into melting analysis probes.</p><p id="abstract">Hybridization probe melting analysis can be complicated by the presence of sequence variation (benign polymorphisms or other mutations) near the targeted mutation. We investigated the use of "masking" probes to differentiate alleles with similar probe melting temperatures. Selected sequence variation was masked by incorporating mismatches (deletion, unmatched nucleotide, or universal base) into hybridization probes at the polymorphic location. Such masking probes create a probe/target mismatch with all possible alleles at the selected polymorphic location. Any allele with additional variation at another site is identified by a lower probe melting temperature than alleles that vary only at the masked position. This "masking technique" was applied to RET protooncogene and HPA6 mutation detection using unlabeled hybridization probes, a saturating dsDNA dye, and high-resolution melting analysis. Masking probes clearly distinguished all targeted mutations from polymorphisms when at least 1 base pair (bp) separated the mutation from the masked variation. We were able to mask polymorphisms immediately adjacent to mutations, except in certain cases, such as those involving single-base deletion probes when both adjacent positions had the same polymorphic nucleotides. The masking probes can also localize mutations to specific codons or nucleotide positions. Masking probes can simplify melting analysis of complex regions and eliminate the need for sequencing.</p></div></section></article></body></html>