<html class="anndoc" data-anndoc-version="2.0" data-origid="16960800" id="16960800" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>16960800</title></head><body><article><section data-type="article"><div class="content"><p id="title">A genomewide single-nucleotide-polymorphism panel with high ancestry information for African American admixture mapping.</p><p id="abstract">Admixture mapping requires a genomewide panel of relatively evenly spaced markers that can distinguish the ancestral origins of chromosomal segments in admixed individuals. Through use of the results of the International HapMap Project and specific selection criteria, the current study has examined the ability of selected single-nucleotide polymorphisms (SNPs) to extract continental ancestry information in African American subjects and to explore parameters for admixture mapping. Genotyping of two linguistically diverse West African populations (Bini and Kanuri Nigerians, who are Niger-Congo [Bantu] and Nilo-Saharan speakers, respectively), European Americans, and African Americans validated a genomewide set of &gt;4,000 SNP ancestry-informative markers with mean and median F(ST) values &gt;0.59 and mean and median Fisher's information content &gt;2.5. This set of SNPs extracted a larger amount of ancestry information in African Americans than previously reported SNP panels and provides nearly uniform coverage of the genome. Moreover, in the current study, simulations show that this more informative panel improves power for admixture mapping in African Americans when ethnicity risk ratios are modest. This is particularly important in the application of admixture mapping in complex genetic diseases for which only modest ethnicity risk ratios of relevant susceptibility genes are expected.</p></div></section></article></body></html>