<html class="anndoc" data-anndoc-version="2.0" data-origid="17621639" id="17621639" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>17621639</title></head><body><article><section data-type="article"><div class="content"><p id="title">Detection of pathogenic gene copy number variations in patients with mental retardation by genomewide oligonucleotide array comparative genomic hybridization.</p><p id="abstract">Genomic imbalance is a major cause of developmental disorders. Microarray-based comparative genomic hybridization (aCGH) has revealed frequent imbalances associated with clinical syndromes, but also a large number of copy number variations (CNVs), which have complicated the interpretation of results. We studied 100 consecutive patients with unexplained mental retardation and a normal karyotype using several platforms of CGH arrays. A genomewide array with 44,290 oligonucleotide probes (OaCGH44K) detected imbalances in 15% of cases studied with sizes ranged from 459 kb to 19 Mb while revealing a small number of CNVs (0.72/individual). Another platform with approximately 240,000 oligonucleotide probes (OaCGH244K) revealed a large number of CNVs (20/individual) in selected cases and their normal parents. We used a comprehensive approach for interpreting the results of aCGH, including consideration of the size, inheritance and gene content of CNVs, and consultation with an online Database of Genomic Variants (DGV) and Online Mendelian Inheritance in Men (OMIM) for information on the genes involved. Our study suggests that genomewide oligonucleotide arrays such as the OaCGH44K platform can be used as a powerful diagnostic tool for detection of genomic imbalances associated with unexplained mental retardation or syndromic autism spectrum disorders. It is interesting to note that a small number of common variants were revealed by OaCGH244K in some study subjects but not in their parents and that some inherited CNVs had altered breakpoints. Further investigations on these alterations may provide useful information for understanding the mechanism of CNVs.</p></div></section></article></body></html>