<html class="anndoc" data-anndoc-version="2.0" data-origid="18449911" id="18449911" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>18449911</title></head><body><article><section data-type="article"><div class="content"><p id="title">Evaluation of in silico splice tools for decision-making in molecular diagnosis.</p><p id="abstract">It appears that all types of genomic nucleotide variations can be deleterious by affecting normal pre-mRNA splicing via disruption/creation of splice site consensus sequences. As it is neither pertinent nor realistic to perform functional testing for all of these variants, it is important to identify those that could lead to a splice defect in order to restrict transcript analyses to the most appropriate cases. Web-based tools designed to provide such predictions are available. We evaluated the performance of six of these tools (Splice Site Prediction by Neural Network [NNSplice], Splice-Site Finder [SSF], MaxEntScan [MES], Automated Splice-Site Analyses [ASSA], Exonic Splicing Enhancer [ESE] Finder, and Relative Enhancer and Silencer Classification by Unanimous Enrichment [RESCUE]-ESE) using 39 unrelated retinoblastoma patients carrying different RB1 variants (31 intronic and eight exonic). These 39 patients were screened for abnormal splicing using puromycin-treated cell lines and the results were compared to the predictions. As expected, 17 variants impacting canonical AG/GT splice sites were correctly predicted as deleterious. A total of 22 variations occurring at loosely defined positions (+/-60 nucleotides from an AG/GT site) led to a splice defect in 19 cases and 16 of them were classified as deleterious by at least one tool (84% sensitivity). In other words, three variants escaped in silico detection and the remaining three were correctly predicted as neutral. Overall our results suggest that a combination of complementary in silico tools is necessary to guide molecular geneticists (balance between the time and cost required by RNA analysis and the risk of missing a deleterious mutation) because the weaknesses of one in silico tool may be overcome by the results of another tool.</p></div></section></article></body></html>