<html class="anndoc" data-anndoc-version="2.0" data-origid="18454449" id="18454449" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>18454449</title></head><body><article><section data-type="article"><div class="content"><p id="title">A meta-analysis of nonsense mutations causing human genetic disease.</p><p id="abstract">Nonsense mutations account for approximately 11% of all described gene lesions causing human inherited disease and approximately 20% of disease-associated single-basepair substitutions affecting gene coding regions. Pathological nonsense mutations resulting in TGA (38.5%), TAG (40.4%), and TAA (21.1%) occur in different proportions to naturally occurring stop codons. Of the 23 different nucleotide substitutions giving rise to nonsense mutations, the most frequent are CGA --&gt; TGA (21%; resulting from methylation-mediated deamination) and CAG --&gt; TAG (19%). The differing nonsense mutation frequencies are largely explicable in terms of variable nucleotide substitution rates such that it is unnecessary to invoke differential translational termination efficiency or differential codon usage. Some genes are characterized by numerous nonsense mutations but relatively few if any missense mutations (e.g., CHM) whereas other genes exhibit many missense mutations but few if any nonsense mutations (e.g., PSEN1). Genes in the latter category have a tendency to encode proteins characterized by multimer formation. Consistent with the operation of a clinical selection bias, genes exhibiting an excess of nonsense mutations are also likely to display an excess of frameshift mutations. Tumor suppressor (TS) genes exhibit a disproportionate number of nonsense mutations while most mutations in oncogenes are missense. A total of 12% of somatic nonsense mutations in TS genes were found to occur recurrently in the hypermutable CpG dinucleotide. In a comparison of somatic and germline mutational spectra for 17 TS genes, approximately 43% of somatic nonsense mutations had counterparts in the germline (rising to 98% for CpG mutations). Finally, the proportion of disease-causing nonsense mutations predicted to elicit nonsense-mediated mRNA decay (NMD) is significantly higher (P=1.56 x 10(-9)) than among nonobserved (potential) nonsense mutations, implying that nonsense mutations that elicit NMD are more likely to come to clinical attention.</p></div></section></article></body></html>