<html class="anndoc" data-anndoc-version="2.0" data-origid="20471002" id="20471002" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>20471002</title></head><body><article><section data-type="article"><div class="content"><p id="title">Pooled association tests for rare variants in exon-resequencing studies.</p><p id="abstract">Deep sequencing will soon generate comprehensive sequence information in large disease samples. Although the power to detect association with an individual rare variant is limited, pooling variants by gene or pathway into a composite test provides an alternative strategy for identifying susceptibility genes. We describe a statistical method for detecting association of multiple rare variants in protein-coding genes with a quantitative or dichotomous trait. The approach is based on the regression of phenotypic values on individuals' genotype scores subject to a variable allele-frequency threshold, incorporating computational predictions of the functional effects of missense variants. Statistical significance is assessed by permutation testing with variable thresholds. We used a rigorous population-genetics simulation framework to evaluate the power of the method, and we applied the method to empirical sequencing data from three disease studies.</p></div></section></article></body></html>