<html class="anndoc" data-anndoc-version="2.0" data-origid="20537299" id="20537299" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>20537299</title></head><body><article><section data-type="article"><div class="content"><p id="title">Principal-component analysis for assessment of population stratification in mitochondrial medical genetics.</p><p id="abstract">Although inherited mitochondrial genetic variation can cause human disease, no validated methods exist for control of confounding due to mitochondrial population stratification (PS). We sought to identify a reliable method for PS assessment in mitochondrial medical genetics. We analyzed mitochondrial SNP data from 1513 European American individuals concomitantly genotyped with the use of a previously validated panel of 144 mitochondrial markers as well as the Affymetrix 6.0 (n = 432), Illumina 610-Quad (n = 458), or Illumina 660 (n = 623) platforms. Additional analyses were performed in 938 participants in the Human Genome Diversity Panel (HGDP) (Illumina 650). We compared the following methods for controlling for PS: haplogroup-stratified analyses, mitochondrial principal-component analysis (PCA), and combined autosomal-mitochondrial PCA. We computed mitochondrial genomic inflation factors (mtGIFs) and test statistics for simulated case-control and continuous phenotypes (10,000 simulations each) with varying degrees of correlation with mitochondrial ancestry. Results were then compared across adjustment methods. We also calculated power for discovery of true associations under each method, using a simulation approach. Mitochondrial PCA recapitulated haplogroup information, but haplogroup-stratified analyses were inferior to mitochondrial PCA in controlling for PS. Correlation between nuclear and mitochondrial principal components (PCs) was very limited. Adjustment for nuclear PCs had no effect on mitochondrial analysis of simulated phenotypes. Mitochondrial PCA performed with the use of data from commercially available genome-wide arrays correlated strongly with PCA performed with the use of an exhaustive mitochondrial marker panel. Finally, we demonstrate, through simulation, no loss in power for detection of true associations with the use of mitochondrial PCA.</p></div></section></article></body></html>