<html class="anndoc" data-anndoc-version="2.0" data-origid="20652909" id="20652909" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>20652909</title></head><body><article><section data-type="article"><div class="content"><p id="title">A custom 148 gene-based resequencing chip and the SNP explorer software: new tools to study antibody deficiency.</p><p id="abstract">Hyper-IgM syndrome and Common Variable Immunodeficiency are heterogeneous disorders characterized by a predisposition to serious infection and impaired or absent neutralizing antibody responses. Although a number of single gene defects have been associated with these immune deficiency disorders, the genetic basis of many cases is not known. To facilitate mutation screening in patients with these syndromes, we have developed a custom 300-kb resequencing array, the Hyper-IgM/CVID chip, which interrogates 1,576 coding exons and intron-exon junction regions from 148 genes implicated in B-cell development and immunoglobulin isotype switching. Genomic DNAs extracted from patients were hybridized to the array using a high-throughput protocol for target sequence amplification, pooling, and hybridization. A Web-based application, SNP Explorer, was developed to directly analyze and visualize the single nucleotide polymorphism (SNP) annotation and for quality filtering. Several mutations in known disease-susceptibility genes such as CD40LG, TNFRSF13B, IKBKG, AICDA, as well as rare nucleotide changes in other genes such as TRAF3IP2, were identified in patient DNA samples and validated by direct sequencing. We conclude that the Hyper-IgM/CVID chip combined with SNP Explorer may provide a cost-effective tool for high-throughput discovery of novel mutations among hundreds of disease-relevant genes in patients with inherited antibody deficiency.</p></div></section></article></body></html>