<html class="anndoc" data-anndoc-version="2.0" data-origid="21412948" id="21412948" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>21412948</title></head><body><article><section data-type="article"><div class="content"><p id="title">SNP uniqueness problem: a proof-of-principle in HapMap SNPs.</p><p id="abstract">SNP-based research strongly affects our biomedical and clinically associated knowledge. Nonunique and false-positive SNP existence in commonly used datasets may thus lead to biased, inaccurate clinically associated conclusions. We designed a computational study to reveal the degree of nonunique/false-positive SNPs in the HapMap dataset. Two sets of SNP flanking sequences were used as queries for BLAT analysis against the human genome. 4.2% and 11.9% of HapMap SNPs align to the genome nonuniquely (long and short, respectively). Furthermore, an average of 7.9% nonunique SNPs are included in common commercial genotyping arrays (according to our designed probes). Nonunique SNPs identified in this study are represented to various degrees in clinically associated databases, stressing the consequence of inaccurate SNP annotation and hence SNP utilization. Unfortunately, our results question some disease-related genotyping analyses, raising a worrisome concern on their validity.</p></div></section></article></body></html>