<html class="anndoc" data-anndoc-version="2.0" data-origid="21922597" id="21922597" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>21922597</title></head><body><article><section data-type="article"><div class="content"><p id="title">Fine-tiling array CGH to improve diagnostics for α- and β-thalassemia rearrangements.</p><p id="abstract">Implementation of multiplex ligation-dependent probe amplification (MLPA) for thalassemia causing deletions has lead to the detection of new rearrangements. Knowledge of the exact breakpoint sequences should give more insight into the molecular mechanisms underlying these rearrangements, and would facilitate the design of gap-PCRs. We have designed a custom fine-tiling array with oligonucleotides covering the complete globin gene clusters. We hybridized 27 DNA samples containing newly identified deletions and nine positive controls. We designed specific primers to amplify relatively short fragments containing the breakpoint sequence and analyzed these by direct sequencing. Results from nine positive controls showed that array comparative genomic hybridization (aCGH) is suitable to detect small and large rearrangements. We were able to locate all breakpoints to a region of approximately 2 kb. We designed breakpoint primers for 22 cases and amplification was successful in 19 cases. For 12 of these, the exact locations of the breakpoints were determined. Seven of these deletions have not been reported before. aCGH is a valuable tool for high-resolution breakpoint characterization. The combination of MLPA and aCGH has lead to relatively cheap and easy to perform PCR assays, which might be of use for laboratories as an alternative for MLPA in populations where only a limited number of specific deletions occur with high frequency.</p></div></section></article></body></html>