<html class="anndoc" data-anndoc-version="2.0" data-origid="23084292" id="23084292" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>23084292</title></head><body><article><section data-type="article"><div class="content"><p id="title">Unraveling multiple MHC gene associations with systemic lupus erythematosus: model choice indicates a role for HLA alleles and non-HLA genes in Europeans.</p><p id="abstract">We have performed a meta-analysis of the major-histocompatibility-complex (MHC) region in systemic lupus erythematosus (SLE) to determine the association with both SNPs and classical human-leukocyte-antigen (HLA) alleles. More specifically, we combined results from six studies and well-known out-of-study control data sets, providing us with 3,701 independent SLE cases and 12,110 independent controls of European ancestry. This study used genotypes for 7,199 SNPs within the MHC region and for classical HLA alleles (typed and imputed). Our results from conditional analysis and model choice with the use of the Bayesian information criterion show that the best model for SLE association includes both classical loci (HLA-DRB1(∗)03:01, HLA-DRB1(∗)08:01, and HLA-DQA1(∗)01:02) and two SNPs, rs8192591 (in class III and upstream of NOTCH4) and rs2246618 (MICB in class I). Our approach was to perform a stepwise search from multiple baseline models deduced from a priori evidence on HLA-DRB1 lupus-associated alleles, a stepwise regression on SNPs alone, and a stepwise regression on HLA alleles. With this approach, we were able to identify a model that was an overwhelmingly better fit to the data than one identified by simple stepwise regression either on SNPs alone (Bayes factor [BF] &gt; 50) or on classical HLA alleles alone (BF &gt; 1,000).</p></div></section></article></body></html>