<html class="anndoc" data-anndoc-version="2.0" data-origid="7668286" id="7668286" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>7668286</title></head><body><article><section data-type="article"><div class="content"><p id="title">Arylamine N-acetyltransferase (NAT2) mutations and their allelic linkage in unrelated Caucasian individuals: correlation with phenotypic activity.</p><p id="abstract">The polymorphic arylamine N-acetyltransferase (NAT2; EC 2.3.1.5) is supposed to be a susceptibility factor for several drug side effects and certain malignancies. A group of 844 unrelated German subjects was genotyped for their acetylation type, and 563 of them were also phenotyped. Seven mutations of the NAT2 gene were evaluated by allele-specific PCR (mutation 341C to T) and PCR-RFLP for mutations at nt positions 191, 282, 481, 590, 803, and 857. From the mutation pattern eight different alleles, including the wild type coding for rapid acetylation and seven alleles coding for slow phenotype, were determined. Four hundred ninety-seven subjects had a genotype of slow acetylation (58.9%; 95% confidence limits 55.5%-62.2%). Phenotypic acetylation capacity was expressed as the ratio of 5-acetylamino-6-formylamino-3-methyluracil and 1-methylxanthine in urine after caffeine intake. Some 6.7% of the cases deviated in genotype and phenotype, but sequencing DNA of these probands revealed no new mutations. Furthermore, linkage pattern of the mutations was always confirmed, as tested in 533 subjects. In vivo acetylation capacity of homozygous wild-type subjects (NAT2*4/*4) was significantly higher than in heterozygous genotypes (P = .001). All mutant alleles showed low in vivo acetylation capacities, including the previously not-yet-defined alleles *5A, *5C, and *13. Moreover, distinct slow genotypes differed significantly among each other, as reflected in lower acetylation capacity of *6A, *7B, and *13 alleles than the group of *5 alleles. The study demonstrated differential phenotypic activity of various NAT2 genes and gives a solid basis for clinical and molecular-epidemiological investigations.</p></div></section></article></body></html>