<html class="anndoc" data-anndoc-version="2.0" data-origid="8352278" id="8352278" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>8352278</title></head><body><article><section data-type="article"><div class="content"><p id="title">Statistical evaluation of multiple-locus linkage data in experimental species and its relevance to human studies: application to nonobese diabetic (NOD) mouse and human insulin-dependent diabetes mellitus (IDDM).</p><p id="abstract">Common, familial human disorders generally do not follow Mendelian inheritance patterns, presumably because multiple loci are involved in disease susceptibility. One approach to mapping genes for such traits in humans is to first study an analogous form in an animal model, such as mouse, by using inbred strains and backcross experiments. Here we describe methodology for analyzing multiple-locus linkage data from such experimental backcrosses, particularly in light of multilocus genetic models, including the effects of epistasis. We illustrate these methods by using data from backcrosses involving nonobese diabetic mouse, which serves as an animal model for human insulin-dependent diabetes mellitus. We show that it is likely that a minimum of nine loci contribute to susceptibility, with strong epistasis effects among these loci. Three of the loci actually confer a protective effect in the homozygote, compared with the heterozygote. Further, we discuss the relevance of these studies for analogous studies of the human form of the trait. Specifically, we show that the magnitude of the gene effect in the experimental backcross is likely to correlate only weakly, at best, with the expected magnitude of effect for a human form, because in humans the gene effect will depend more heavily on disease allele frequencies than on the observed penetrance ratios; such allele frequencies are unpredictable. Hence, the major benefit from animal studies may be a better understanding of the disease process itself, rather than identification of cells through comparison mapping in humans by using regions of homology.</p></div></section></article></body></html>