<html class="anndoc" data-anndoc-version="2.0" data-origid="9311748" id="9311748" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>9311748</title></head><body><article><section data-type="article"><div class="content"><p id="title">Accurate inference of relationships in sib-pair linkage studies.</p><p id="abstract">Relative-pair designs are routinely employed in linkage studies of complex genetic diseases and quantitative traits. Valid application of these methods requires correct specification of the relationships of the pairs. For example, within a sibship, presumed full sibs actually might be MZ twins, half sibs, or unrelated. Misclassification of half-sib pairs or unrelated individuals as full sibs can result in reduced power to detect linkage. When other family members, such as parents or additional siblings, are available, incorrectly specified relationships usually will be detected through apparent incompatibilities with Mendelian inheritance. Without other family members, sibling relationships cannot be determined absolutely, but they still can be inferred probabilistically if sufficient genetic marker data are available. In this paper, we describe a simple likelihood ratio method to infer the true relationship of a putative sibling pair. We explore the number of markers required to accurately infer relationships typically encountered in a sib-pair study, as a function of marker allele frequencies, marker spacing, and genotyping error rate, and we conclude that very accurate inference of relationships can be achieved, given the marker data from even part of a genome scan. We compare our method to related methods of relationship inference that have been suggested. Finally, we demonstrate the value of excluding non-full sibs in a genetic linkage study of non-insulin-dependent diabetes mellitus.</p></div></section></article></body></html>