<html class="anndoc" data-anndoc-version="2.0" data-origid="25141179" id="25141179" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>25141179</title></head><body><article><section data-type="article"><div class="content"><p id="s1p1">Saturation editing of genomic regions by multiplex homology-directed repair.</p><p id="s1p2">Saturation mutagenesis--coupled to an appropriate biological assay--represents a fundamental means of achieving a high-resolution understanding of regulatory and protein-coding nucleic acid sequences of interest. However, mutagenized sequences introduced in trans on episomes or via random or "safe-harbour" integration fail to capture the native context of the endogenous chromosomal locus. This shortcoming markedly limits the interpretability of the resulting measurements of mutational impact. Here, we couple CRISPR/Cas9 RNA-guided cleavage with multiplex homology-directed repair using a complex library of donor templates to demonstrate saturation editing of genomic regions. In exon 18 of BRCA1, we replace a six-base-pair (bp) genomic region with all possible hexamers, or the full exon with all possible single nucleotide variants (SNVs), and measure strong effects on transcript abundance attributable to nonsense-mediated decay and exonic splicing elements. We similarly perform saturation genome editing of a well-conserved coding region of an essential gene, DBR1, and measure relative effects on growth that correlate with functional impact. Measurement of the functional consequences of large numbers of mutations with saturation genome editing will potentially facilitate high-resolution functional dissection of both cis-regulatory elements and trans-acting factors, as well as the interpretation of variants of uncertain significance observed in clinical sequencing.</p></div></section></article></body></html>