<html class="anndoc" data-anndoc-version="2.0" data-origid="24092741" id="24092741" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>24092741</title></head><body><article><section data-type="article"><div class="content"><p id="s1p1">Following gene duplication, paralog interference constrains transcriptional circuit evolution.</p><p id="s1p2">Most models of gene duplication assume that the ancestral functions of the preduplication gene are independent and can therefore be neatly partitioned between descendant paralogs. However, many gene products, such as transcriptional regulators, are components within cooperative assemblies; here, we show that a natural consequence of duplication and divergence of such proteins can be competitive interference between the paralogs. Our example is based on the duplication of the essential MADS-box transcriptional regulator Mcm1, which is found in all fungi and regulates a large set of genes. We show that a set of historical amino acid sequence substitutions minimized paralog interference in contemporary species and, in doing so, increased the molecular complexity of this gene regulatory network. We propose that paralog interference is a common constraint on gene duplicate evolution, and its resolution, which can generate additional regulatory complexity, is needed to stabilize duplicated genes in the genome.</p></div></section></article></body></html>