<html class="anndoc" data-anndoc-version="2.0" data-origid="7473737" id="7473737" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>7473737</title></head><body><article><section data-type="article"><div class="content"><p id="title">Mutations at positions 153 and 328 in Escherichia coli alkaline phosphatase provide insight towards the structure and function of mammalian and yeast alkaline phosphatases.</p><p id="abstract">In order to understand some of the differences between human placental, human, Saccharomyces cerevisiae and Escherichia coli alkaline phosphatases in specific activity, activation by magnesium, and pH versus activity profiles, the X-ray crystal structures of three mutant E. coli alkaline phosphatases have been determined. The aligned sequences of alkaline phosphatases from mammalian, yeast and E. coli show that 25 to 30% of the amino acids are absolutely conserved and the active site residues are completely conserved with the exception of residues 153, 328 and 155. The bacterial enzyme has a salt-bridge, Asp153/Lys328, near the third metal binding site which, based on sequence homology, is apparently absent in the yeast and mammalian enzymes. The human enzymes have histidine at positions 153 and 328, and the yeast enzyme has histidine at position 328. In the E. coli enzyme, Asp153 was replaced by histidine (D153H), Lys328 was replaced by histidine (K328H), and a double mutant (DM) was constructed containing both mutations. The structure of the K328H enzyme was refined using cross-validation to a resolution of 2.3 A with a working R-factor of 0.181 and a free R-factor of 0.249. The DM structure was determined to a resolution of 2.5 A with a working R-factor of 0.166 and a free R-factor of 0.233. The structure of the D135H enzyme, which has been reported to a resolution of 2.4 A, has been re-refined using cross-validation to a working R-factor of 0.179 and a free R-factor of 0.239 for controlled comparisons with the two new structures. In all three structures the most significant changes are related to the bound phosphate inhibitor and the identity of the metal ion in the third binding site. The changes in the position of the phosphate group and the alterations at the third metal binding site indicate the structural basis for the variations in the steady-state kinetic parameters previously reported for these enzymes.</p></div></section></article></body></html>