<html class="anndoc" data-anndoc-version="2.0" data-origid="9865957" id="9865957" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>9865957</title></head><body><article><section data-type="article"><div class="content"><p id="title">Probing the ligand binding domain of the GluR2 receptor by proteolysis and deletion mutagenesis defines domain boundaries and yields a crystallizable construct.</p><p id="abstract">Ionotropic glutamate receptors constitute an important family of ligand-gated ion channels for which there is little biochemical or structural data. Here we probe the domain structure and boundaries of the ligand binding domain of the AMPA-sensitive GluR2 receptor by limited proteolysis and deletion mutagenesis. To identify the proteolytic fragments, Maldi mass spectrometry and N-terminal amino acid sequencing were employed. Trypsin digestion of HS1S2 (Chen GQ, Gouaux E. 1997. Proc Natl Acad Sci USA 94:13431-13436) in the presence and absence of glutamate showed that the ligand stabilized the S1 and S2 fragments against complete digestion. Using limited proteolysis and multiple sequence alignments of glutamate receptors as guides, nine constructs were made, folded, and screened for ligand binding activity. From this screen, the S1S21 construct proved to be trypsin- and chymotrypsin-resistant, stable to storage at 4 degrees C, and amenable to three-dimensional crystal formation. The HS1S21 variant was readily prepared on a large scale, the His tag was easily removed by trypsin, and crystals were produced that diffracted to beyond 1.5 A resolution. These experiments, for the first time, pave the way to economical overproduction of the ligand binding domains of glutamate receptors and more accurately map the boundaries of the ligand binding domain.</p></div></section></article></body></html>