<html class="anndoc" data-anndoc-version="2.0" data-origid="24390348" id="24390348" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>24390348</title></head><body><article><section data-type="article"><div class="content"><p id="s1p1">Landscape of genomic alterations in cervical carcinomas.</p><p id="s1p2">Cervical cancer is responsible for 10-15% of cancer-related deaths in women worldwide. The aetiological role of infection with high-risk human papilloma viruses (HPVs) in cervical carcinomas is well established. Previous studies have also implicated somatic mutations in PIK3CA, PTEN, TP53, STK11 and KRAS as well as several copy-number alterations in the pathogenesis of cervical carcinomas. Here we report whole-exome sequencing analysis of 115 cervical carcinoma-normal paired samples, transcriptome sequencing of 79 cases and whole-genome sequencing of 14 tumour-normal pairs. Previously unknown somatic mutations in 79 primary squamous cell carcinomas include recurrent E322K substitutions in the MAPK1 gene (8%), inactivating mutations in the HLA-B gene (9%), and mutations in EP300 (16%), FBXW7 (15%), NFE2L2 (4%), TP53 (5%) and ERBB2 (6%). We also observe somatic ELF3 (13%) and CBFB (8%) mutations in 24 adenocarcinomas. Squamous cell carcinomas have higher frequencies of somatic nucleotide substitutions occurring at cytosines preceded by thymines (Tp*C sites) than adenocarcinomas. Gene expression levels at HPV integration sites were statistically significantly higher in tumours with HPV integration compared with expression of the same genes in tumours without viral integration at the same site. These data demonstrate several recurrent genomic alterations in cervical carcinomas that suggest new strategies to combat this disease.</p></div></section></article></body></html>