<html class="anndoc" data-anndoc-version="2.0" data-origid="8589684" id="8589684" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>8589684</title></head><body><article><section data-type="article"><div class="content"><p id="s1p1">Three new adenosine deaminase mutations that define a splicing enhancer and cause severe and partial phenotypes: implications for evolution of a CpG hotspot and expression of a transduced ADA cDNA.</p><p id="s1p2">We report three novel adenosine deaminase (ADA) mutations with interesting implications. A Somali child with severe combined immunodeficiency disease (SCID) had reduced ADA mRNA in T cells and was homozygous for the nonsense mutation Q3X. Unexpectedly, her healthy father was a compound ADA heterozygote whose second allele carried a 'partial' mutation, R142Q, due to a G--&gt;A transition of a CpG dinucleotide. A C--&gt;T transition of the same CpG produced a nonsense mutation, R142X, in two homozygous Canadian Mennonite infants with SCID. The severe and healthy phenotypes associated with R142X and R142Q, the high frequency of 'partial' ADA mutations arising from CpGs in healthy individuals of African descent and the presence of CAA (glutamine) at codon 142 in murine ADA, suggest selection for replacement of this CpG hotspot by CpA during ADA evolution. R142X, located within a purine-rich segment at nt 62/116 of exon 5, caused skipping of the exon, possibly by disrupting a splicing enhancer. Absence of exon 5 in T cell ADA mRNA and low ADA activity in T cells and erythrocytes obtained at age 18-22 months from one of the Mennonite children, indicate limited expression of a normal ADA cDNA from retrovirally transduced CD34+ umbilical cord leukocytes infused shortly after birth in an attempt at stem cell gene therapy.</p></div></section></article></body></html>