<html class="anndoc" data-anndoc-version="2.0" data-origid="8554073" id="8554073" lang="" xml:lang="" xmlns="http://www.w3.org/1999/xhtml"><head><title>8554073</title></head><body><article><section data-type="article"><div class="content"><p id="s1p1">Mutation analysis of very-long-chain acyl-coenzyme A dehydrogenase (VLCAD) deficiency: identification and characterization of mutant VLCAD cDNAs from four patients.</p><p id="s1p2">Very-long-chain acyl-coenzyme A dehydrogenase (VLCAD) deficiency is a newly identified disease. A 105-bp deletion in the VLCAD cDNA in two patients has been reported, and detailed molecular characterization of this disease has remained to be done. We report here five mutations identified in four patients: a 135-bp deletion encompassing bases 343-477, a C-1837-to-T transition (R613W), 3-bp deletions at the nucleotide positions 388-390 (E130del) and 895-897 (K299del), and an A-1144-to-C transversion (K382Q). Sequencing of genomic DNA amplified by PCR revealed a 135-bp deletion caused by exon skipping due to a 1-bp deletion in a 3' splice site of an intron. In cDNA expression experiments using Chinese hamster ovary (CHO) cells, we found that each of the mRNAs derived from E130del and K299del clones were unstable and that translation products from R613W, E130del, K299del, and K382Q clones were labile. Each of R613W, E130del, K299del, and K382Q proteins expressed in CHO cells appeared abnormal in dimer assembly, as shown in gel-filtration analysis. VLCAD activity was not detected in mutants' transfectants. Thus, we verified that all five mutations identified in these four patients were disease-causing alterations.</p></div></section></article></body></html>